WcaOnRails/app/models/competition.rb

```diff
@@ -458,11 +465,82 @@ def in_progress?
   end
 
   def is_over?
-    end_date < Date.today
+    !!end_date && end_date < Date.today
+  end
+
+  def results_updated_at
+    Option.from_nilable(results.order('updated_at desc').pluck(:updated_at).first)
```

Comment:
Ahh, I see. But won't a `.pluck` cause a select of every single result and then generate a ruby array of their `updated_at` fields? That feels worse than selecting every column of exactly 1 row. See:

```
irb(main):004:0> r = Result.pluck(:id); nil
   (1.4ms)  SELECT `Results`.`id` FROM `Results`
D, [2016-04-15T02:30:42.953473 #12480] DEBUG -- :    (1.4ms)  SELECT `Results`.`id` FROM `Results`
=> nil
irb(main):006:0> r = Result.first.id; nil
  Result Load (0.4ms)  SELECT  `Results`.* FROM `Results`  ORDER BY `Results`.`id` ASC LIMIT 1
D, [2016-04-15T02:31:01.408759 #12480] DEBUG -- :   Result Load (0.4ms)  SELECT  `Results`.* FROM `Results`  ORDER BY `Results`.`id` ASC LIMIT 1
=> nil
```

Perhaps the best of both worlds?

```
irb(main):005:0> r = Result.limit(1).pluck(:id); nil
   (0.3ms)  SELECT  `Results`.`id` FROM `Results` LIMIT 1
D, [2016-04-15T02:30:53.736865 #12480] DEBUG -- :    (0.3ms)  SELECT  `Results`.`id` FROM `Results` LIMIT 1
=> nil
```